File: ios/FoodTruckApp/FoodTruckApp/AppConfig.swift
import Foundation

struct AppConfig {
    // MARK: - Environment Configuration
    static let isDevelopment: Bool = {
        #if DEBUG
        return false // 프로덕션 서버 사용
        #else
        return false
        #endif
    }()
    
    // MARK: - Server Configuration
    private static let devLocalIP = "192.168.219.200"
    private static let devPort = "3002"
    private static let prodBaseURL = "https://truck.carrera74.com"
    
    // MARK: - URLs
    static var serverURL: String {
        return isDevelopment ? "http://\(devLocalIP):\(devPort)" : prodBaseURL
    }
    
    static var apiURL: String {
        return "\(serverURL)/api"
    }
    
    static var mobileURL: String {
        return "\(serverURL)/mobile"
    }
    
    // MARK: - Environment Info
    static var environment: String {
        return isDevelopment ? "development" : "production"
    }
    
    // MARK: - Debug Info
    static var debugInfo: String {
        return """
        Environment: \(environment)
        Server URL: \(serverURL)
        API URL: \(apiURL)
        Mobile URL: \(mobileURL)
        """
    }
    
    // MARK: - App Info
    static let appName = "세종유미곱창트럭"
    static let appVersion = "1.0.0"
    static let bundleIdentifier = "com.foodtruck.app"
    
    // MARK: - Firebase Configuration
    static let firebaseConfigFile = "GoogleService-Info"
    
    // MARK: - Notification Configuration
    static let notificationChannelId = "foodtruck_notifications"
    static let notificationChannelName = "푸드트럭 알림"
    static let notificationChannelDescription = "푸드트럭의 메뉴, 위치 정보 알림"
}

File: android/app/src/main/java/com/foodtruck/app/config/AppConfig.kt
package com.foodtruck.app.config

import com.foodtruck.app.BuildConfig

object AppConfig {
    // 환경 설정 (BuildConfig에서 가져옴)
    private val IS_DEVELOPMENT = BuildConfig.IS_DEVELOPMENT
    
    // 서버 URL 설정 (BuildConfig에서 가져옴)
    private val DEV_LOCAL_IP = BuildConfig.DEV_LOCAL_IP
    private val DEV_PORT = BuildConfig.DEV_PORT
    private val PROD_BASE_URL = BuildConfig.PROD_BASE_URL
    
    // 현재 환경에 따른 서버 URL 반환
    fun getServerUrl(): String {
        return if (IS_DEVELOPMENT) {
            "http://$DEV_LOCAL_IP:$DEV_PORT"
        } else {
            PROD_BASE_URL
        }
    }
    
    // API 엔드포인트 URL
    fun getApiUrl(): String = "${getServerUrl()}/api"
    
    // 모바일 웹 URL
    fun getMobileUrl(): String = "${getServerUrl()}/mobile"
    
    // 환경 정보
    fun isDevelopment(): Boolean = IS_DEVELOPMENT
    
    fun getEnvironment(): String = if (IS_DEVELOPMENT) "development" else "production"
    
    // 디버그 정보
    fun getDebugInfo(): String {
        return """
            Environment: ${getEnvironment()}
            Server URL: ${getServerUrl()}
            API URL: ${getApiUrl()}
            Mobile URL: ${getMobileUrl()}
        """.trimIndent()
    }
}
